silhouette/app/com/mohiva/play/silhouette/api/Silhouette.scala

```diff
@@ -266,10 +267,7 @@ trait Silhouette[I <: Identity, A <: Authenticator] extends Controller with Logg
     private def handleInitializedAuthenticator[T](authenticator: A, block: A => Future[HandlerResult[T]])(implicit request: RequestHeader) = {
       val auth = env.authenticatorService.touch(authenticator)
       block(auth.extract).flatMap {
-        case hr @ HandlerResult(pr: Authenticator.Discard, _) =>
-          env.authenticatorService.discard(authenticator, pr).map(pr => hr.copy(pr))
-        case hr @ HandlerResult(pr: Authenticator.Renew, _) =>
-          env.authenticatorService.renew(authenticator, pr).map(pr => hr.copy(pr))
+        case hr @ HandlerResult(pr: AuthenticatorResult, _) => Future.successful(hr)
         case hr @ HandlerResult(pr, _) => auth match {
```

Comment:
As pointed out, `HandlerResult` has nothing to do with authenticators. This is a different beast and it's needed for the [request handlers](http://silhouette.mohiva.com/v2.0/docs/endpoints#request-handlers). 

But were do you see Discard & Renew classes?

> The code should directly update the Result....there is no reason to delay it using the Update & Discard markers any more

What do you mean here?